ambuild2/context.py

```diff
@@ -92,17 +93,34 @@ def reconfigure(self):
 
         # The database should be upgraded here, so we should always have an
         # API version set.
-        api_version = self.db.query_var('api_version')
+        api_version = Version(self.db.query_var('api_version'))
         assert api_version is not None
 
-        if api_version == '2.0':
-            from ambuild2.frontend.v2_0.amb2.gen import Generator
-        elif api_version == '2.1':
-            from ambuild2.frontend.v2_1.amb2 import Generator
-
-        gen = Generator.FromVars(self.vars, self.db, self.options.refactor)
+        from ambuild2.frontend.amb2 import Generator
+
+        if api_version >= '2.1':
+            from ambuild2.frontend.v2_1.context_manager import ContextManager
+        elif api_version >= '2.0':
+            from ambuild2.frontend.v2_0.context_manager import ContextManager
+
+        # Backwards compatibility: for an automatic reconfigure on an older build,
+        # just assume the source path is the cwd. If the AMBuildScript suddenly
+        # has decided to depend on originalCwd, then the user may have to manually
+        # run configure.py again, until we remove configure.py entirely.
+        if 'originalCwd' in self.vars:
+            originalCwd = self.vars['originalCwd']
+        else:
+            originalCwd = self.vars['sourcePath']
```

Comment:
'self.vars' is just a ```dict``` no? If so, this is simpler to read is it not?
```suggestion
        originalCwd = self.vars.get('originalCwd', self.vars['sourcePath'])
```